### PR TITLE
Implement the database connection and fetch announcements on request

### PR DESCRIPTION
**Pull request by leomet07**
Create the /api/get_announcements route (that caches the mongodb connection so that connections are not created for every request)

Also, the mongoDB connection (in the global namespace declaration) has proper types (added in global.d.ts) 

Add type support for the ReceievedAnnouncement

On every request to / (the homepage), request the /api/get_announcements route and fetch the latest announcements (implemented through getServerSideProps)

For this request, the server that is pinged will dynamically switch from either using the development localhost server or the URL that Vercel assigns its server (whatever that may be, production, development, or preview)

### CODE REVIEW

File: pages/api/get_announcements.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+					return newv;
+				}) as ReceivedAnnouncement[];
+
+				res.status(200).json({ success: true, data: announcements });
+			} catch (error) {
+				console.error(error);
+				res.status(400).json({ success: false });
+			}
+			break;
+		// case "POST":
```
Comment (TheEgghead27) [inline pages/api/get_announcements.ts]: Perhaps it would be prudent to purge this commented out block of code.

File: models/Announcement.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import mongoose from "mongoose";
+
+const AnnouncementSchema = new mongoose.Schema({
+	title: String,
```
Comment (TheEgghead27) [inline models/Announcement.ts:4]: I was under the impression that there would also be a date string in the announcement, unless I am mistaken

File: pages/index.tsx [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
+import { Key } from "react";
 
-const Home: NextPage = () => {
+const Home = (
+	props: InferGetServerSidePropsType<typeof getServerSideProps>
+) => {
+	const parseDate = function (d: string) {
+		return new Date(d).toLocaleString().split(",")[0];
+	};
+	console.log("Server url", props.serverUrl);
```
Comment (TheEgghead27) [inline pages/index.tsx:19]: Perhaps this debug log is no longer necessary

**Review by TheEgghead27 — CHANGES_REQUESTED** [review]
There's nothing necessarily blocking about my comments, but it would be helpful to have them addressed (or explicitly ignored) before merging

File: pages/index.tsx (hunk 3 above)
Comment (leomet07) [reply pages/index.tsx:19]: Let’s keep it in.
It’s going to be helpful with all of the vercel preview branches.

File: models/Announcement.ts (hunk 2 above)
Comment (leomet07) [reply models/Announcement.ts:4]: It’s not stored in the database under an explicit field, it’s implicitly stored in the ObjectId.
The date is extracted AFTER fetching from the DB, server side (see get_announcements.ts)

File: pages/api/get_announcements.ts (hunk 1 above)
Comment (leomet07) [reply pages/api/get_announcements.ts]: Agreed, TODO.

**Review by TheEgghead27 — APPROVED** [review]

**Pull request merged**